### PR TITLE
Fixed compilation errors under Archlinux and possibly other platforms

### LINKED ISSUE #68: cross-compilation in openwrt fails
This came up during a test build for openwrt #62 but it possibly hints at something of wider significance.
Unfortunately, github is stupid and you can't even attach a txt file (seriously? Hello?!) so I will link the build log to a pastebin here that will expire in 90 days. http://paste.debian.net/164080/

### PR DESCRIPTION
**Pull request by zoopp**
* Removed redundant inequality comparison operator from asiotap::endpoint. This
  operator introduces overload resolution ambiguity and on top of that it is
  already implemented by boost::variant;
* Fixed a compilation issue triggered by comparing a boost::variant to a foreign
  type.

Possibly fixes #68.

### CODE REVIEW

**Comment by ereOn** [comment]
@zoopp Thanks for this ! Look good to me.

I updated and fixed the continuous integration on the master branch. Could you please rebase your work on top of the latest green build so I can merge ?

**Comment by zoopp** [comment]
Turns out that if we want to compile against both boost 1.55 and the latest stable (1.58) we have to use the more uglier form inside the if comparison.

**Comment by ereOn** [comment]
@zoopp If the operator is not used in the core codebase but is in the samples, then the samples are testing something useless. They are called "samples" but really are intended to be tests (crappy/inconvenient ones if you ask me) I would just adapt the sample to not use the operator in this case.

As for the boost 1.55/1.58 changes : if the only change to support boost 1.55 is to change a couple of statements to use the "uglier" form, I think it is an acceptable trade-off.

Long story short, I'd say, use your best judgement: If one way or the other makes your life easier and doesn't cause any drawback for other builds/platforms, go for it.

**Comment by ereOn** [comment]
@zoopp Any news on that front ? Do you need any help ?

It seems that your fix would also fix #86.

**Comment by zoopp** [comment]
Sorry, I've been quite busy lately. I've been meaning to finish this during this weekend.

**Comment by ereOn** [comment]
@zoopp Awesome ! Thanks.

**Comment by zoopp** [comment]
Alright, the build passes. In the end I decided to conditionally include the target operator based on the boost version we're compiling against.

**Comment by ereOn** [comment]
Thanks a lot @zoopp !

**Comment by leggewie** [comment]
Are we going to see a new release pretty soon?  2.0 now fails in Debian Unstable because of this issue.  Backporting this fix then makes for a successful compile but releasing 2.1 seems preferable.

@ereOn, what do you think?

**Comment by ereOn** [comment]
@leggewie I'm all for another release. I will need some time to do it though.

**Comment by leggewie** [comment]
only 3 bugs left marked as [blocking 2.1](https://github.com/freelan-developers/freelan/milestone/4) ;-)
